Assets/PatchKit Patcher/Scripts/UI/Changelog/ChangelogList.cs

```
@@ -20,15 +23,73 @@ protected override IEnumerator LoadCoroutine()
                 yield return null;
             }
 
+            var appSecret = Patcher.Instance.Data.Value.AppSecret;
+
+            LoadChangelogFromCache(appSecret);
+
             yield return
-                Threading.StartThreadCoroutine(() => MainApiConnection.GetAppVersionList(Patcher.Instance.Data.Value.AppSecret, null, CancellationToken.Empty),
-                    response =>
-                    {
-                        foreach (var version in response.OrderByDescending(version => version.Id))
-                        {
-                            CreateVersionChangelog(version);
-                        }
-                    });
+                Threading.StartThreadCoroutine(() =>
+                        MainApiConnection.GetAppVersionList(
+                            Patcher.Instance.Data.Value.AppSecret,
+                            null,
+                            CancellationToken.Empty),
+                    versions => CreateAndCacheChangelog(appSecret, versions));
+        }
+
+        private void LoadChangelogFromCache(string appSecret)
+        {
+            try
+            {
+                var cacheValue = new UnityCache(appSecret).GetValue("app-changelog", null);
+
+                if (cacheValue == null)
+                {
+                    return;
+                }
+
+                var versions = JsonConvert.DeserializeObject<AppVersion[]>(cacheValue);
+
+                CreateChangelog(versions);
+            }
+            catch (Exception)
+            {
+                // ignore
+            }
+        }
+
+        private void CreateAndCacheChangelog(string appSecret, AppVersion[] versions)
+        {
+            try
+            {
+                var cacheValue = JsonConvert.SerializeObject(versions);
+
+                new UnityCache(appSecret).SetValue("app-changelog", cacheValue);
+            }
+            catch (Exception)
+            {
+                // ignore
```

Instead of completely ignoring exceptions, please make a log file entry.